Library/Formula/z3.rb

```diff
@@ -12,31 +19,88 @@ class Z3 < Formula
   if build.without?("python3") && build.without?("python")
     odie "z3: --with-python3 must be specified when using --without-python"
   end
-
-  bottle do
-    cellar :any
-    sha256 "ea169ccefdbebdd17213b4fab603dce2029b03bde0b62fa98920cbaf431d4771" => :el_capitan
-    sha256 "e8f726245f283d43efe68f2516ebf1fc62fd2ab486a850befc0c388ef9f5c1ed" => :yosemite
-    sha256 "2c67f6d604e3b478bac87e223891f3252a8f29a048564a91a8ea57f6c3b9a8ba" => :mavericks
-  end
+  depends_on :java => :optional
+  depends_on "ocaml" => :optional
 
   def install
     inreplace "scripts/mk_util.py", "dist-packages", "site-packages"
 
-    Language::Python.each_python(build) do |python, version|
-      system python, "scripts/mk_make.py", "--prefix=#{prefix}"
+    # Installing  JARs to "lib" can cause conflicts between packages. Therefore install the jar to libexec
+    inreplace "scripts/mk_util.py", "out.write('\\t@cp %s.jar %s.jar\\n' % (self.package_name, os.path.join('$(PREFIX)', 'lib', self.package_name)))", "out.write('\\t@cp %s.jar %s.jar\\n' % (self.package_name, os.path.join('$(PREFIX)', 'libexec', self.package_name)))"
```

Comment:
We need changes like this to be accepted upstream to include this into Homebrew.